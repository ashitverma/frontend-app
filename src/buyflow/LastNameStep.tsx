import React, { useState } from "react";

interface LastNameStepProps {
  cb: (field: string, value: string) => void;
}

const LastNameStep: React.FC<LastNameStepProps> = (props) => {
  const [lastName, setLastName] = useState("");
  return (
    <>
      <div>
        <label htmlFor="lastName">Last Name : </label>
        <input
          type="text"
          onChange={({ target: { value } }) => {
            setLastName(value);
          }}
          value={lastName}
        ></input>
      </div>
      <button onClick={() => props.cb("lastName", lastName)}>Next</button>
    </>
  );
};

export default LastNameStep;
