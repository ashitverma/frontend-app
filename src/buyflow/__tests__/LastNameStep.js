import { render } from "@testing-library/react"; // (or /dom, /vue, ...)
import LastNameStep from "../LastNameStep";

test("renders a lastName step input field", () => {
  const queries = render(<LastNameStep />);
  expect(queries.getByText(/last name/i)).toBeInTheDocument();
});

it("should take a snapshot", () => {
  const { asFragment } = render(<LastNameStep />);

  expect(asFragment(<LastNameStep />)).toMatchSnapshot();
});
