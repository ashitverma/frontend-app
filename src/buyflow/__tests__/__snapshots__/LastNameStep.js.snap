// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      for="lastName"
    >
      Last Name:
    </label>
    <input
      type="text"
      value=""
    />
  </div>
  <button>
    Next
  </button>
</DocumentFragment>
`;
